mage/main.go

```diff
@@ -468,35 +469,54 @@ type mainfileTemplateData struct {
 	BinaryName  string
 }
 
-func listGoFiles(magePath, goCmd, tags string, env []string) ([]string, error) {
-	args := []string{"list"}
-	if tags != "" {
-		args = append(args, fmt.Sprintf("-tags=%s", tags))
-	}
-	args = append(args, "-e", "-f", `{{join .GoFiles "||"}}`)
-	cmd := exec.Command(goCmd, args...)
-	cmd.Env = env
-	buf := &bytes.Buffer{}
-	cmd.Stderr = buf
-	cmd.Dir = magePath
-	b, err := cmd.Output()
+// listGoFiles returns a list of all .go files in a given directory,
+// matching the provided tag
+func listGoFiles(magePath, goCmd, tag string, envStr []string) ([]string, error) {
+	origMagePath := magePath
+	if !filepath.IsAbs(magePath) {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return nil, err
```

Comment:
everywhere we return an error, we should be prepending some kind of additional info onto it, so that if it bubbles up to the user, they get useful debugging info.

In this case, something like
```suggestion
			return nil, fmt.Errorf("can't get current working directory: %v", err)
```
Assume the level above will add the bit about the fact that it was calling listGoFiles.

(note that we use %v instead of %w, so it'll still compile with older versions of go)